src/interface/swarm.cpp

```diff
@@ -237,6 +239,30 @@ void Swarm::setPoolMax(const std::int64_t nmax_pool) {
   }
 
   nmax_pool_ = nmax_pool;
+
+  // Eliminate any cached SwarmPacks, as they will need to be rebuilt following setPoolMax
+  // (1): MeshData "base" partitions
+  for (int i = 0; i < pm->DefaultNumPartitions(); i++) {
+    auto pmdp = pm->mesh_data.GetOrAdd("base", i);
+    auto &real_mdp_cache = pmdp->GetSwarmPackCache<Real>().pack_map;
+    auto &int_mdp_cache = pmdp->GetSwarmPackCache<int>().pack_map;
+    if (real_mdp_cache.size() > 0) real_mdp_cache.clear();
+    if (int_mdp_cache.size() > 0) int_mdp_cache.clear();
+  }
+
+  // (2): MeshData "base"
+  auto pmd = pm->mesh_data.Get("base");
+  auto &real_md_cache = pmd->GetSwarmPackCache<Real>().pack_map;
+  auto &int_md_cache = pmd->GetSwarmPackCache<int>().pack_map;
+  if (real_md_cache.size() > 0) real_md_cache.clear();
+  if (int_md_cache.size() > 0) int_md_cache.clear();
+
+  // (3): MeshBlockData
+  auto pmbd = pmb->meshblock_data.Get().get();
+  auto &real_mbd_cache = pmbd->GetSwarmPackCache<Real>().pack_map;
+  auto &int_mbd_cache = pmbd->GetSwarmPackCache<int>().pack_map;
+  if (real_mbd_cache.size() > 0) real_mbd_cache.clear();
+  if (int_mbd_cache.size() > 0) int_mbd_cache.clear();
```

Comment:
This feels like it should live in `MeshData` and `MeshBlockData`, i.e. there should be a function `ClearSwarmCaches` in those that just gets called here.

Reply:
Addressed in [59c0d1c](https://github.com/parthenon-hpc-lab/parthenon/pull/1037/commits/59c0d1c3118de6fa5d40af115c818d1c5c67b419)